apps/designer/app/canvas/shared/styles.ts

```diff
@@ -250,6 +272,37 @@ const useUpdateStyle = () => {
       }
       utils.tree.setInstanceStyleMutable(rootInstance, id, updates, breakpoint);
     });
+
+    store.createTransaction([stylesContainer], (styles) => {
+      const instanceId = id;
+      const breakpointId = breakpoint.id;
+      for (const update of updates) {
+        const matchedIndex = styles.findIndex(
```

Comment:
could be a useful reusable styles utility, no?

Reply:
You mean findIndex? We may end up with some kind of indexes to speedup access. But it's a different story.

Reply:
no, the whole block

Reply:
It's gonna be used only here. Where do you want to reuse it?